main.go

```diff
@@ -116,6 +116,10 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "Glance")
 		os.Exit(1)
 	}
+	if err = (&glancev1.Glance{}).SetupWebhookWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create webhook", "webhook", "Glance")
+		os.Exit(1)
+	}
```

Comment:
Should we wrap this in `if strings.ToLower(os.Getenv("ENABLE_WEBHOOKS")) != "false" {` like you had for the other PR?  Or did you perhaps figure out a way to get the webhooks to run locally?

Reply:
I can add that, but not sure I understand what you mean by locally.  Do you mean webhook as a different local process? It works for me  when the operator is deployed on crc. I just looked at https://book.kubebuilder.io/cronjob-tutorial/running-webhook.html. Also, I don't know how you can set the env variable to enable disable the webhook.

Reply:
By locally I mean at the CLI via (in your `glance-operator` dir): `ENABLE_WEBHOOKS=false GOWORK= OPERATOR_TEMPLATES=./templates make run`.  This can be useful for developmental iterating and not having to redeploy the operator via OLM, etc, when you want to test changes.

Reply:
I think it works fine if you generate the self-signed certs. We can add a target to generate self-signed certs to Makefile.

```
[stack@osp-df-2 glance-operator]$ openssl req -newkey rsa:4096 -days 3650 -nodes -x509   -keyout /tmp/k8s-webhook-server/serving-certs/tls.key   -out /tmp/k8s-webhook-server/serving-certs/tls.crt
....+...+.....+...+....+...+..+.+......+...........+.........................+..+.+.....+......+...+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.+...+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*..+..................+.........+.+............+..............+.+.........+...+........+...+.........+...+...............+...+.+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
.+....................+......+.........+....+..............+...+.........+.........+.+......+..+...+.......+...+...+.....+.+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*................+.....+..........+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*....+......+......+...+.....+.+............+..+...+...+......................+.................+...+...+....+..+....+.........+....................+.+.........+.....+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-----
You are about to be asked to enter information that will be incorporated
into your certificate request.
What you are about to enter is what is called a Distinguished Name or a DN.
There are quite a few fields but you can leave some blank
For some fields there will be a default value,
If you enter '.', the field will be left blank.
-----
Country Name (2 letter code) [XX]:IN
State or Province Name (full name) []:MH
Locality Name (eg, city) [Default City]:PNQ
Organization Name (eg, company) [Default Company Ltd]:Red Hat
Organizational Unit Name (eg, section) []:DF
Common Name (eg, your name or your server's hostname) []:osp-df2
Email Address []:
[stack@osp-df-2 glance-operator]$ GOWORK= OPERATOR_TEMPLATES=./templates make run
/home/stack/work/glance-operator/bin/controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases && \
rm -f api/bases/* && cp -a config/crd/bases api/
/home/stack/work/glance-operator/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
test -f go.work || go work init
go work use .
go work use ./api
go vet ./...
go vet ./... ./api/...
go run ./main.go
I0131 07:35:39.771728 1634555 request.go:682] Waited for 1.038209601s due to client-side throttling, not priority and fairness, request: GET:https://api.crc.testing:6443/apis/security.openshift.io/v1?timeout=32s
1.6751685414748962e+09	INFO	controller-runtime.metrics	Metrics server is starting to listen	{"addr": ":8080"}
1.675168541478722e+09	INFO	controller-runtime.builder	skip registering a mutating webhook, object does not implement admission.Defaulter or WithDefaulter wasn't called	{"GVK": "glance.openstack.org/v1beta1, Kind=Glance"}
1.6751685414787903e+09	INFO	controller-runtime.builder	Registering a validating webhook	{"GVK": "glance.openstack.org/v1beta1, Kind=Glance", "path": "/validate-glance-openstack-org-v1beta1-glance"}
1.6751685414788954e+09	INFO	controller-runtime.webhook	Registering webhook	{"path": "/validate-glance-openstack-org-v1beta1-glance"}
1.6751685414790215e+09	INFO	setup	starting manager
1.675168541479233e+09	INFO	controller-runtime.webhook.webhooks	Starting webhook server
1.6751685414793482e+09	INFO	Starting server	{"kind": "health probe", "addr": "[::]:8081"}
1.675168541479361e+09	INFO	Starting server	{"path": "/metrics", "kind": "metrics", "addr": "[::]:8080"}
1.6751685414794888e+09	INFO	Starting EventSource	{"controller": "glanceapi", "controllerGroup": "glance.openstack.org", "controllerKind": "GlanceAPI", "source": "kind source: *v1beta1.GlanceAPI"}
1.6751685414794989e+09	INFO	Starting EventSource	{"controller": "glance", "controllerGroup": "glance.openstack.org", "controllerKind": "Glance", "source": "kind source: *v1beta1.Glance"}
1.675168541479546e+09	INFO	Starting EventSource	{"controller": "glanceapi", "controllerGroup": "glance.openstack.org", "controllerKind": "GlanceAPI", "source": "kind source: *v1beta1.KeystoneEndpoint"}
1.6751685414795575e+09	INFO	Starting EventSource	{"controller": "glance", "controllerGroup": "glance.openstack.org", "controllerKind": "Glance", "source": "kind source: *v1beta1.GlanceAPI"}
1.6751685414795642e+09	INFO	Starting EventSource	{"controller": "glanceapi", "controllerGroup": "glance.openstack.org", "controllerKind": "GlanceAPI", "source": "kind source: *v1.Service"}
1.6751685414795775e+09	INFO	Starting EventSource	{"controller": "glance", "controllerGroup": "glance.openstack.org", "controllerKind": "Glance", "source": "kind source: *v1beta1.MariaDBDatabase"}
1.675168541479585e+09	INFO	Starting EventSource	{"controller": "glanceapi", "controllerGroup": "glance.openstack.org", "controllerKind": "GlanceAPI", "source": "kind source: *v1.Secret"}
1.6751685414795997e+09	INFO	Starting EventSource	{"controller": "glanceapi", "controllerGroup": "glance.openstack.org", "controllerKind": "GlanceAPI", "source": "kind source: *v1.ConfigMap"}
1.6751685414795897e+09	INFO	Starting EventSource	{"controller": "glance", "controllerGroup": "glance.openstack.org", "controllerKind": "Glance", "source": "kind source: *v1beta1.KeystoneService"}
1.6751685414796495e+09	INFO	Starting EventSource	{"controller": "glance", "controllerGroup": "glance.openstack.org", "controllerKind": "Glance", "source": "kind source: *v1.PersistentVolumeClaim"}
1.6751685414796588e+09	INFO	controller-runtime.certwatcher	Updated current TLS certificate
1.6751685414796715e+09	INFO	Starting EventSource	{"controller": "glance", "controllerGroup": "glance.openstack.org", "controllerKind": "Glance", "source": "kind source: *v1.Job"}
1.6751685414796872e+09	INFO	Starting EventSource	{"controller": "glance", "controllerGroup": "glance.openstack.org", "controllerKind": "Glance", "source": "kind source: *v1.ConfigMap"}
1.6751685414796984e+09	INFO	Starting EventSource	{"controller": "glance", "controllerGroup": "glance.openstack.org", "controllerKind": "Glance", "source": "kind source: *v1.Secret"}
1.6751685414796138e+09	INFO	Starting EventSource	{"controller": "glanceapi", "controllerGroup": "glance.openstack.org", "controllerKind": "GlanceAPI", "source": "kind source: *v1.Deployment"}
1.6751685414797342e+09	INFO	Starting EventSource	{"controller": "glanceapi", "controllerGroup": "glance.openstack.org", "controllerKind": "GlanceAPI", "source": "kind source: *v1.Route"}
1.675168541479744e+09	INFO	Starting Controller	{"controller": "glanceapi", "controllerGroup": "glance.openstack.org", "controllerKind": "GlanceAPI"}
1.6751685414797142e+09	INFO	Starting Controller	{"controller": "glance", "controllerGroup": "glance.openstack.org", "controllerKind": "Glance"}
1.675168541479816e+09	INFO	controller-runtime.certwatcher	Starting certificate watcher
1.675168541479833e+09	INFO	controller-runtime.webhook	Serving webhook server	{"host": "", "port": 9443}

```

Reply:
That's awesome!  Let's get that Makefile target in place before we merge any of these PRs with webhooks, since without the `ENABLE_WEBHOOKS` check we would otherwise be breaking the local `make run` for them.

Reply:
Added target in the latest patchset.